### DISCUSSION TITLE
A generic method to compare Top Level List in Java

### DISCUSSION THREAD

**Discussion by guileluk** [javers/javers#1173] [Q&A]
I would like to create a generic method to compare Top Level List 

example :  
public Diff compareListMethod (List<?> beforeObject, List<?> afterObject , Class<?> class) {
return javers.compareCollections(beforeObject, afterObject , class);
}

Due to the requirement, the class parameter I passed in coming from Class.forname("<theClassNameString>"), since I need to make it dynamic.
I got error from the above, it is not accept Class<?> 

Can anyone tell me how to make it work ?

Many Thanks

**Comment by bartoszwalacik** [comment]
sorry, but I don't understand your question. You can pass any class instance to `compareCollections()`. Where is the issue?

**Reply by guileluk** [reply]
The question is , I can not pass the class instance.
You can copy and paste the method in my query, and then you will see the compilations error. 

The scenario is : 
I had a DB table, which had 2 fields.
1st field is a json field which stored a json which representing different types of object.
2nd field is a string which stored the Classname which I want the json converted to. 

e.g. 
1st
[{
name: myname,
age: 30
}
]
2nd field
com.Person

1st
[{
DeptName: deptname,
members: 40
}
]
2nd field
com.Department

**Reply by bartoszwalacik** [reply]
If you don't know what kind of objects are stored in your collections, simply pass Object.class
